--- /etc/init.d/S35iptables.orig
+++ /etc/init.d/S35iptables
@@ -7,6 +7,7 @@
 start() {
 	printf 'Starting %s: ' "$DAEMON"
 	iptables-restore /etc/iptables.conf
+	[ -f /etc/ip6tables.conf ] && ip6tables-restore /etc/ip6tables.conf
 	status=$?
 	if [ "$status" -eq 0 ]; then
 		echo "OK"
@@ -19,6 +20,7 @@
 stop() {
 	printf 'Stopping %s: ' "$DAEMON"
 	iptables -F
+	ip6tables -F
 	status=$?
 	if [ "$status" -eq 0 ]; then
 		echo "OK"
@@ -37,6 +39,7 @@
 save() {
 	printf 'Saving %s: ' "$DAEMON"
 	iptables-save -f /etc/iptables.conf
+	ip6tables-save -f /etc/ip6tables.conf
 	status=$?
 	if [ "$status" -eq 0 ]; then
 		echo "OK"
